### PR TITLE
fix: python layer tests

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
* fix python layer tests
* changed order to save layer state then cfn templates

#### Description of how you validated changes
* updated tests for the changes

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/execute-provider-utils.test.ts [hunk 1]
```diff
@@ -11,7 +12,7 @@ jest.mock('../../../../__mocks__/faked-plugin', () => ({
   },
 }));
 describe('executeProviderUtils', () => {
-  const mockContext = {};
+  const mockContext: $TSAny = {};
```
Comment (jhockett) [inline packages/amplify-cli/src/__tests__/extensions/amplify-helpers/execute-provider-utils.test.ts:15]: please revert these changes, it will conflict with https://github.com/aws-amplify/amplify-cli/pull/7153

**Review by jhockett — CHANGES_REQUESTED** [review]

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/execute-provider-utils.test.ts (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-cli/src/__tests__/extensions/amplify-helpers/execute-provider-utils.test.ts]: done, thanks for catching this

**Review by jhockett — APPROVED** [review]
